HDS-3124 Select doc update (#2041)
hashicorp/design-system: website/docs/components/form/select/partials/guidelines/guidelines.md

!!! Info

**This component uses a native HTML element**

> It's not possible to create custom variations of this component as it uses a native HTML element. Find more information [on MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select).

Reviewer: Should we be more specific, and explain that it's not possible to create custom variations of the `<option>` elements?

Author: Isn't this also applicable to the `<select>` element too, where we can't apply non-native HTML customizations to it?

Author: Initially when updating this my intent was to make it seem that any "custom variation" (which to your point Jory may be not accurate enough) within the select component (meant to be technical variation) was not possible and intentionally kept it vague so that it encompasses more than just the `<option>`.

But to your point, maybe we need to say "any technical custom variation" isn't allowed? I Want to include more than the option list so that this is "future" proof and that no one comes and says we didn't explicitly say the `<select>` element wasn't mentioned.

Author: Ok wording was updated again.

> **This component is a native HTML element**
> 
> Creating custom non-native variations of this component is not possible as we rely on the brower to render it as-is. More information can be found [on MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select).

Reviewer 2: Great addition! I found the original content to be easier to parse and more accurate (since the visual stylings aren't rendered "as-is"). Also updated the second sentence to use active voice instead of passive voice. 

```suggestion
**This component uses a native HTML element**

It's not possible to create custom variations of this component as it uses a native HTML element. Find more information [on MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select).
```

> It's not possible to customize options within this component as it is a native HTML element, and uses browser defaults to render it. Find more information [on MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select).

Reviewer 3: I'm not sure if the comma is needed here

```suggestion
It's not possible to customize options within this component as it is a native HTML element and uses browser defaults to render it. Find more information [on MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select).
```
